test/Domain/LeanCode.CQRS.Tests/Execution/QueryExecutionTests.cs

```diff
@@ -42,6 +46,30 @@ public async Task Correctly_routes_data_through_pipeline_elements()
             Assert.Equal(query, element.Data);
         }
 
+        [Fact]
+        public async Task Correctly_log_query_response_in_pipeline_element()
```

Comment:
This is not the correct place to test this - you should have a separate test class for `ResponseLoggerElement` and `DevResponseLoggerElement` that tests just that. This class tests `QueryExecutionTests` and it has nothing to do with these two elements. Let's move that there. And also - we don't need whole pipeline here, we're just testing the single element so we can test it directly.

Reply:
Same applies to the `OperationExecutionTests`.